contrib/bdii/htcondor-ce-provider

```diff
@@ -221,14 +234,22 @@ def main():
         running_vo_jobs = collections.defaultdict(int)
         total_vo_jobs = collections.defaultdict(int)
 
-        for job in query:
-            if not job.get("JobStatus") or not job.get("x509userproxyvoname"):
-                continue
-            total_vo_jobs[job['x509userproxyvoname']] += 1
-            if job['JobStatus'] == 1:
-                idle_vo_jobs[job['x509userproxyvoname']] += 1
-            elif job['JobStatus'] == 2:
-                running_vo_jobs[job['x509userproxyvoname']] += 1
+        signal.signal(signal.SIGALRM, handler)
+        signal.alarm(10)
+        try:
+            for job in query:
+                if not job.get("JobStatus") or not job.get("x509userproxyvoname"):
+                    continue
+                total_vo_jobs[job['x509userproxyvoname']] += 1
+                if job['JobStatus'] == 1:
+                    idle_vo_jobs[job['x509userproxyvoname']] += 1
+                elif job['JobStatus'] == 2:
+                    running_vo_jobs[job['x509userproxyvoname']] += 1
+        except Exception, e:
```

Comment:
Same thing about catching `RuntimeError` or `TimeoutException`

Reply:
There is no timeout on this query. If the sched was slow this would have been discovered with the earlier ping.